### PR TITLE
Docs: Rearrange rules for better categories (and improve rule summaries)

### PR DESCRIPTION
**Pull request by brettz9**

### CODE REVIEW

File: docs/rules/README.md [hunk 1]
```diff
@@ -139,9 +138,9 @@ These rules are specific to JavaScript running on Node.js or using CommonJS in t
 * [no-mixed-requires](no-mixed-requires.md) - disallow mixing regular variable and require declarations
 * [no-new-require](no-new-require.md) - disallow use of `new` operator with the `require` function
 * [no-path-concat](no-path-concat.md) - disallow string concatenation with `__dirname` and `__filename`
+* [no-process-env](no-process-env.md) - disallow use of `process.env`
 * [no-process-exit](no-process-exit.md) - disallow `process.exit()`
-* [no-restricted-imports](no-restricted-imports.md) - restrict usage of specified Node.js imports
-* [no-restricted-modules](no-restricted-modules.md) - restrict usage of specified Node.js modules
+* [no-restricted-modules](no-restricted-modules.md) - restrict usage of specified modules when loaded by `require` function [see https://nodejs.org/api/globals.html#globals_require and https://nodejs.org/api/modules.html#modules_addenda_package_manager_tips]
```
Comment (pedrottimark) [inline docs/rules/README.md]: @brettz9 uh oh 😱 my bad to include the cross-references in square brackets. which were for your benefit in a way that looked as if they belonged in the list

File: docs/rules/README.md [hunk 2]
```diff
@@ -231,6 +230,7 @@ These rules are only relevant to ES6 environments.
 * [no-const-assign](no-const-assign.md) - disallow modifying variables that are declared using `const` (recommended)
 * [no-dupe-class-members](no-dupe-class-members.md) - disallow duplicate name in class members (recommended)
 * [no-new-symbol](no-new-symbol.md) - disallow use of the `new` operator with the `Symbol` object (recommended)
+* [no-restricted-imports](no-restricted-imports.md) - restrict usage of specified modules when loaded by `import` declaration [see http://www.ecma-international.org/ecma-262/6.0/index.html#sec-imports and https://github.com/eslint/eslint/blob/master/lib/rules/no-restricted-imports.js#L22]
```
Comment (pedrottimark) [inline docs/rules/README.md]: @brettz9 uh oh 😱 my bad to include the cross-references in square brackets. which were for your benefit in a way that looked as if they belonged in the list

**Comment by brettz9** [comment]
No worries--amended.

File: docs/rules/README.md (hunk 2 above)
Comment (brettz9) [inline docs/rules/README.md]: Not sure if I need to reply directly under the line notes, especially for outdated diffs, but I added a note indicating this was amended.

**Comment by ilyavolodin** [comment]
LGTM. Thanks for PR!

**Pull request merged**